internal/compiler-interface/src/main/java/xsbti/VirtualFileRef.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Zinc - The incremental compiler for Scala.
+ * Copyright Lightbend, Inc. and Mark Harrah
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package xsbti;
+
+/*
+ * Represents a reference to a file-like object.
```

Comment:
How does one dereference to get to the underlying content or the underlying physical file? Why is a reference required? Some of these design notes could live here.

How are `VirtualFileRef`-s  expected to behave for equality -- is it just based on `.id`?

Reply:
Added docs in https://github.com/sbt/zinc/pull/712/commits/5137b10ef029bf4d2500e191677122e0cfdbe766